src/iterator.cr

```diff
@@ -200,6 +205,58 @@ module Iterator(T)
     end
   end
 
+  # The same as `#chain`, but have better performance when the quantity of
+  # iterators to chain is large(usually greater than 4) or undetermined.
+  # It uses lesser type specification for more freedom to avoid the failure (at a cost of performance),
+  # and a loop rather than recursive calling to achieve better performance for numbers of iterators
+  # (but when the quantity is small, it's not really efficiency).
```

Comment:
We don't tend to discuss implementation details in docs, only user-visible effects, like performance and ability to chain in a loop.